queries/fortran/highlights.scm

```diff
@@ -1,114 +1,274 @@
-(identifier) @variable
-(string_literal) @string
-(number_literal) @number
-(boolean_literal) @boolean
-(comment) @comment
+; Preprocs
+
+(preproc_file_line) @preproc
+
+; Namespaces
+
+(program_statement
+  (name) @namespace)
+
+(end_program_statement
+  (name) @namespace)
+
+(module_statement
+  (name) @namespace)
+
+(end_module_statement
+  (name) @namespace)
+
+(submodule_statement
+  (name) @namespace)
+
+(end_submodule_statement
+  (name) @namespace)
+
+; Includes
 
 [
-  (intrinsic_type)
-  "pointer"
+  "import"
+  "include"
+  "use"
+] @include
+
+(import_statement
+  ","
+  ["all" "none"] @keyword)
+
+; Attributes
+
+[
+  (none)
+  "implicit"
+  "intent"
+] @attribute
+
+(implicit_statement
+  "type" @attribute)
+
+; Keywords
+
+[
+  "attributes"
+  "associate"
+  "block"
+  "class"
+  "contains"
+  "default"
+  "dimension"
+  "endassociate"
+  "endenum"
+  "endinterface"
+  "endmodule"
+  "endselect"
+  "endsubmodule"
+  "enum"
+  "enumerator"
+  "equivalence"
+  "extends"
+  "goto"
+  "interface"
+  "intrinsic"
+  "non_intrinsic"
+  "module"
+  "submodule"
+  "namelist"
+  "parameter"
+  "quiet"
+  "rank"
+  "save"
+  "selectcase"
+  "selectrank"
+  "selecttype"
+  "sequence"
+  "stop"
+  "target"
+] @keyword
+
+[
+  (default)
+] @keyword
+
+; Types
+
+[
+  (type_name)
 ] @type
 
 [
+  (intrinsic_type)
+] @type.builtin
+
+; Typedefs
+
+[
+  "classis"
+  "endtype"
+  "type"
+  "typeis"
+] @type.definition
```

Comment:
These should be `@keyword`. The ever-misleading `@type.definition` capture is for type aliases.